Add 1-arg, 2-arg, 3-arg methods

iamdanfox opened the pull request:
This is hopefully pre-empting a guava 21 style breakage, where a change could be compilation compatible but not binary compatible.

uschi2000 reviewed (approved):
Can you add the rationale in the commit message, please?

uschi2000 commented:
oh, I see you did already. maybe elaborate a little.

gracew commented:
is the idea that we may care about the perf of allocating arrays, and don't want to get into a compilation-compatible but binary-incompatible situation later when adding these methods?

iamdanfox commented:
@gracew yep I think so, it's basically just that if we don't opt-in now, we likely don't really have a way to opt-in to these methods later without a full breaking release..

iamdanfox committed:
Add 1-arg, 2-arg, 3-arg methods

These methods avoid allocating an array, so should have a tiny
performance improvement on the happy path.

Adding them now to hopefully preempt a guava 21 style breakage, because
while adding these later would be compilation compatible, the resultant
binary would not be compatible!

iamdanfox commented:
Have updated the commit message. Gonna tag an rc so we can try this out!
